tests/foreman/api/test_contentviewfilter.py

```diff
@@ -764,3 +798,181 @@ def test_positive_search_rpm(self):
+        cls.product = entities.Product(organization=cls.org).create()
+        cls.repo = entities.Repository(
+                content_type='yum',
+                product=cls.product,
+                url=CUSTOM_MODULE_STREAM_REPO_2).create()
+        cls.repo.sync()
+
+    def setUp(self):
+        """Init content view with repo per each test"""
+        super(ContentViewFilterRuleTestCase, self).setUp()
+        self.content_view = entities.ContentView(
+            organization=self.org,
+        ).create()
+        self.content_view.repository = [self.repo]
+        self.content_view.update(['repository'])
+
+    @tier2
+    def test_positive_promote_module_stream_filter(self):
+        """Verify Module Stream, Errata Count after Promote, Publish for Content View
+        with Module Stream Exclude Filter
+
+        :id: 2f5d21b1-8cbc-4a77-b8a2-09aa466f56a3
+
+        :expectedresults: Content View should get published and promoted successfully
+            with correct Module Stream count.
+
+        :CaseLevel: Integration
+        """
+        # Exclude module stream filter
+        cv_filter = entities.ModuleStreamContentViewFilter(content_view=self.content_view,
+                                                           inclusion=False).create()
+        module_streams = entities.ModuleStream().search(
+            query=dict(search='name="{}"'.format('duck')))
+        entities.ContentViewFilterRule(content_view_filter=cv_filter,
+                                       module_stream=module_streams).create()
+        self.content_view.publish()
+        content_view = self.content_view.read()
+        content_view_version_info = content_view.version[0].read()
+        self.assertEqual(len(content_view.repository), 1)
+        self.assertEqual(len(content_view.version), 1)
+
+        #  the module stream and errata count based in filter after publish
+        self.assertEqual(content_view_version_info.module_stream_count, 4)
+        self.assertEqual(content_view_version_info.errata_counts['total'], 3)
+
+        # Promote Content View
+        lce = entities.LifecycleEnvironment(organization=self.org).create()
+        promote(content_view.version[0], lce.id)
+        content_view = content_view.read()
+        content_view_version_info = content_view.version[0].read()
+
+        # assert the module stream and errata count based in filter after promote
+        self.assertEqual(content_view_version_info.module_stream_count, 4)
+        self.assertEqual(content_view_version_info.errata_counts['total'], 3)
+
+    @tier2
+    def test_positive_include_exclude_module_stream_filter(self):
+        """Verify Include and Exclude Errata filter(modular errata) automatically force the copy
+           of the module streams associated to it.
+
+        :id: 20540722-b163-4ebb-b18d-351444ef0c86
+
+        :steps:
+            1. Create Include Errata filter with 'RHEA-2012:0059' has (duck and kangaroo streams)
```

Comment:
I mentioned this because, if any new person reviews the test case could easily understand. If you think it is not useful I will update you as suggested (will make more generic).